internal/codegen/tpl_file.go

```diff
@@ -271,6 +271,70 @@ func (f *TplFile) MigrateTo(path string) (out string, err error) {
 	return "", nil
 }
 
+// MigrateFrom parses blocks from the provided file as if they
+// originated from this template. This is useful when you want to move a
+// file, but keep its contents. This should be called before any other
+// functions are called inside of the file.
+//
+// Note: Only blocks are persisted. Unlike [MigrateTo] the file is not
+// actually copied to disk.
+//
+// **Example `old.go`:**
+//
+//	package main
+//
+//	import "fmt"
+//
+//	## <<Stencil::Block(main-function)>>
+//	func main() {
+//	    fmt.Println("Hello from migrated code!")
+//	}
+//	## <</Stencil::Block>>
+//
+// **Example `new.go.tpl`:**
+//
+//	{{ file.MigrateFrom "old.go" }}
+//	package main
+//
+//	import "fmt"
+//
+//	## <<Stencil::Block(main-function)>>
+//	{{ file.Block "main-function" }}
+//	## <</Stencil::Block>>
+//
+// **Example `new.go`:**
+//
+//	package main
+//
+//	import "fmt"
+//
+//	## <<Stencil::Block(main-function)>>
+//	func main() {
+//	    fmt.Println("Hello from migrated code!")
+//	}
+//	## <</Stencil::Block>>
+//
+// Result: The block content from old.go is preserved in new.go
+func (f *TplFile) MigrateFrom(path string) (out string, err error) {
+	blocks, err := parseBlocks(path, f.t)
```

Comment:
check if file exists first?  it should noop for future runs with migratefrom

Reply:
lol, yep, good catch. Will add a test case for it as well